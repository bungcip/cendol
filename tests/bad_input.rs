use cendol::test_utils::compile_and_get_error;
use insta::assert_yaml_snapshot;

#[test]
fn test_unexpected_token() {
    let err = compile_and_get_error("int main() { int a = 1 +; }", "unexpected_token.c");
    assert_yaml_snapshot!("unexpected_token", err.unwrap_err());
}

#[test]
fn test_type_mismatch() {
    let err = compile_and_get_error("int main() { int a = \"hello\"; }", "type_mismatch.c");
    assert_yaml_snapshot!("type_mismatch", err.unwrap());
}

#[test]
fn test_undefined_label() {
    let err = compile_and_get_error(
        "int main() { goto undefined_label; return 0; }",
        "undefined_label.c",
    );
    assert_yaml_snapshot!("undefined_label", err.unwrap_err());
}

#[test]
fn test_assignment_to_const() {
    let err = compile_and_get_error(
        "int main() { const int x = 10; x = 20; return x; }",
        "assignment_to_const.c",
    );
    assert_yaml_snapshot!("assignment_to_const", err.unwrap_err());
}
