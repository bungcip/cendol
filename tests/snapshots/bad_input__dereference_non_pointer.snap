---
source: tests/bad_input.rs
expression: err.unwrap_err()
---
msg: "Cannot dereference non-pointer type `Int`"
path: dereference_non_pointer.c
span:
  file: 0
  start:
    file: 0
    line: 1
    column: 27
  end:
    file: 0
    line: 1
    column: 28
verbose: false
kind: Error
